src/Analyser/MutatingScope.php

```diff
@@ -1021,12 +1021,7 @@ private function resolveType(Expr $node): Type
 			}
 
 			if ($type instanceof IntegerRangeType) {
-				$negativeRange = $this->resolveType(new Node\Expr\BinaryOp\Mul($node->expr, new LNumber(-1)));
-
-				if ( $negativeRange instanceof IntegerRangeType && ($negativeRange->getMin() === null || $negativeRange->getMax() === null)) {
```

Comment:
This check was moved into the integerRangeMath method and therefore fixes multiplication with a negative number for the general case